### ISSUE TITLE
PHP Notice: Undefined variable: username in C:\inetpub\glpi\src\Reservation.php on line 559

### ISSUE THREAD

**Issue by eduardomozart** [glpi-project/glpi#13353]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.5

### Bug description

After upgrading to GLPI 10.0.5 from 9.2.4 -> 9.4.6 -> 10.0.5, the following notice is being shown a lot at PHP error log file (``php_errors.log``).

Also noticed that the system is being reported as "Windows Server 2016" at the GLPI setup information section but it's a Windows Server 2019 instance.

### Relevant log output

```shell
[21-Nov-2022 08:27:50 America/So_Paulo] PHP Notice: Undefined variable: username in C:\inetpub\glpi\src\Reservation.php on line 559
```

### Your GLPI setup information

````
<details><summary>Instruções de instalação e configuração</summary><pre>GLPI 10.0.5 ( => C:\inetpub\glpi)
Installation mode: TARBALL
Current language:pt_BR

</details><details><summary>Server</summary><pre> 
Operating system: Windows NT UNIFI 10.0 build 17763 (Windows Server 2016) AMD64
PHP 7.4.33 cgi-fcgi (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apcu, bcmath, bz2, calendar, cgi-fcgi, ctype,
	curl, date, dom, exif, fileinfo, filter, gd, hash, iconv, imap, intl, json, ldap, libxml, mbstring, mysqli, mysqlnd, openssl,
	pcre, readline, session, sodium, standard, tokenizer, xml, xmlreader, xmlwriter, zip, zlib)
Setup: max_execution_time="600" memory_limit="256M" post_max_size="8M" safe_mode="" session.save_handler="files"
	upload_max_filesize="20M" 
Software: Microsoft-IIS/10.0
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36
Server Software: MySQL Community Server (GPL)
	Server Version: 5.7.40-log
	Server SQL Mode: STRICT_TRANS_TABLES,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
	Parameters: [Omitted]
	Host info: localhost via TCP/IP
	
PHP version (7.4.33) is supported.
Sessions configuration is OK.
Allocated memory is sufficient.
mysqli extension is installed.
Following extensions are installed: dom, fileinfo, json, simplexml.
curl extension is installed.
gd extension is installed.
intl extension is installed.
libxml extension is installed.
zlib extension is installed.
The constant SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES is present.
Database engine version (5.7.40) is supported.
No files from previous GLPI version detected.
The log file has been created successfully.
Write access to C:\inetpub\glpi/files/_cache has been validated.
Write access to C:\inetpub\glpi/config has been validated.
Write access to C:\inetpub\glpi/files/_cron has been validated.
Write access to C:\inetpub\glpi/files has been validated.
Write access to C:\inetpub\glpi/files/_dumps has been validated.
Write access to C:\inetpub\glpi/files/_graphs has been validated.
Write access to C:\inetpub\glpi/files/_lock has been validated.
Write access to C:\inetpub\glpi/files/_pictures has been validated.
Write access to C:\inetpub\glpi/files/_plugins has been validated.
Write access to C:\inetpub\glpi/files/_rss has been validated.
Write access to C:\inetpub\glpi/files/_sessions has been validated.
Write access to C:\inetpub\glpi/files/_tmp has been validated.
Write access to C:\inetpub\glpi/files/_uploads has been validated.
The following directories should be placed outside "C:\inetpub\glpi":
‣ "C:\inetpub\glpi/files" ("GLPI_VAR_DIR")
‣ "C:\inetpub\glpi\config" ("GLPI_CONFIG_DIR")
You can ignore this suggestion if you are certain that these directories are not accessible through your web server.
Sessions configuration is secured.
exif extension is installed.
ldap extension is installed.
openssl extension is installed.
zip extension is installed.
bz2 extension is installed.
Zend OPcache extension is installed.
Following extensions are installed: ctype, iconv, mbstring, sodium.
Write access to C:\inetpub\glpi/marketplace has been validated.
Timezones seems loaded in database.

</details><details><summary>GLPI constants</summary><pre> 
GLPI_ROOT: "C:\\inetpub\\glpi"
GLPI_CONFIG_DIR: "C:\\inetpub\\glpi/config"
GLPI_VAR_DIR: "C:\\inetpub\\glpi/files"
GLPI_MARKETPLACE_DIR: "C:\\inetpub\\glpi/marketplace"
GLPI_USE_CSRF_CHECK: "1"
GLPI_CSRF_EXPIRES: "7200"
GLPI_CSRF_MAX_TOKENS: "100"
GLPI_USE_IDOR_CHECK: "1"
GLPI_IDOR_EXPIRES: "7200"
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: false
GLPI_SERVERSIDE_URL_ALLOWLIST: ["/^(https?|feed):\\/\\/[^@:]+(\\/.*)?$/"]
GLPI_TELEMETRY_URI: "https://telemetry.glpi-project.org"
GLPI_INSTALL_MODE: "TARBALL"
GLPI_NETWORK_MAIL: "glpi@teclib.com"
GLPI_NETWORK_SERVICES: "https://services.glpi-network.com"
GLPI_MARKETPLACE_ALLOW_OVERRIDE: true
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: true
GLPI_USER_AGENT_EXTRA_COMMENTS: ""
GLPI_DISABLE_ONLY_FULL_GROUP_BY_SQL_MODE: "1"
GLPI_AJAX_DASHBOARD: "1"
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: "0"
GLPI_CENTRAL_WARNINGS: "1"
GLPI_DOC_DIR: "C:\\inetpub\\glpi/files"
GLPI_CACHE_DIR: "C:\\inetpub\\glpi/files/_cache"
GLPI_CRON_DIR: "C:\\inetpub\\glpi/files/_cron"
GLPI_DUMP_DIR: "C:\\inetpub\\glpi/files/_dumps"
GLPI_GRAPH_DIR: "C:\\inetpub\\glpi/files/_graphs"
GLPI_LOCAL_I18N_DIR: "C:\\inetpub\\glpi/files/_locales"
GLPI_LOCK_DIR: "C:\\inetpub\\glpi/files/_lock"
GLPI_LOG_DIR: "C:\\inetpub\\glpi/files/_log"
GLPI_PICTURE_DIR: "C:\\inetpub\\glpi/files/_pictures"
GLPI_PLUGIN_DOC_DIR: "C:\\inetpub\\glpi/files/_plugins"
GLPI_RSS_DIR: "C:\\inetpub\\glpi/files/_rss"
GLPI_SESSION_DIR: "C:\\inetpub\\glpi/files/_sessions"
GLPI_TMP_DIR: "C:\\inetpub\\glpi/files/_tmp"
GLPI_UPLOAD_DIR: "C:\\inetpub\\glpi/files/_uploads"
GLPI_INVENTORY_DIR: "C:\\inetpub\\glpi/files/_inventories"
GLPI_NETWORK_REGISTRATION_API_URL: "https://services.glpi-network.com/api/registration/"
GLPI_MARKETPLACE_PLUGINS_API_URI: "https://services.glpi-network.com/api/glpi-plugins/"
GLPI_I18N_DIR: "C:\\inetpub\\glpi/locales"
GLPI_VERSION: "10.0.5"
GLPI_SCHEMA_VERSION: "10.0.5@628dbfbb91eb4caf10c35969d9162b9300b141e0"
GLPI_MARKETPLACE_PRERELEASES: false
GLPI_MIN_PHP: "7.4.0"
GLPI_MAX_PHP: "8.3.0"
GLPI_YEAR: "2022"

</details><details><summary>Libraries</summary><pre> 
htmlawed/htmlawed version 1.2.9 in (C:\inetpub\glpi\vendor\htmlawed\htmlawed)
phpmailer/phpmailer version 6.6.0 in (C:\inetpub\glpi\vendor\phpmailer\phpmailer\src)
simplepie/simplepie version 1.5.8 in (C:\inetpub\glpi\vendor\simplepie\simplepie\library)
tecnickcom/tcpdf version 6.5.0 in (C:\inetpub\glpi\vendor\tecnickcom\tcpdf)
michelf/php-markdown in (C:\inetpub\glpi\vendor\michelf\php-markdown\Michelf)
true/punycode in (C:\inetpub\glpi\vendor\true\punycode\src)
iamcal/lib_autolink in (C:\inetpub\glpi\vendor\iamcal\lib_autolink)
sabre/dav in (C:\inetpub\glpi\vendor\sabre\dav\lib\DAV)
sabre/http in (C:\inetpub\glpi\vendor\sabre\http\lib)
sabre/uri in (C:\inetpub\glpi\vendor\sabre\uri\lib)
sabre/vobject in (C:\inetpub\glpi\vendor\sabre\vobject\lib)
laminas/laminas-i18n in (C:\inetpub\glpi\vendor\laminas\laminas-i18n\src)
laminas/laminas-servicemanager in (C:\inetpub\glpi\vendor\laminas\laminas-servicemanager\src)
monolog/monolog in (C:\inetpub\glpi\vendor\monolog\monolog\src\Monolog)
sebastian/diff in (C:\inetpub\glpi\vendor\sebastian\diff\src)
donatj/phpuseragentparser in (C:\inetpub\glpi\vendor\donatj\phpuseragentparser\src\UserAgent)
elvanto/litemoji in (C:\inetpub\glpi\vendor\elvanto\litemoji\src)
symfony/console in (C:\inetpub\glpi\vendor\symfony\console)
scssphp/scssphp in (C:\inetpub\glpi\vendor\scssphp\scssphp\src)
laminas/laminas-mail in (C:\inetpub\glpi\vendor\laminas\laminas-mail\src\Protocol)
laminas/laminas-mime in (C:\inetpub\glpi\vendor\laminas\laminas-mime\src)
rlanvin/php-rrule in (C:\inetpub\glpi\vendor\rlanvin\php-rrule\src)
blueimp/jquery-file-upload in (C:\inetpub\glpi\vendor\blueimp\jquery-file-upload\server\php)
ramsey/uuid in (C:\inetpub\glpi\vendor\ramsey\uuid\src)
psr/log in (C:\inetpub\glpi\vendor\psr\log\Psr\Log)
psr/simple-cache in (C:\inetpub\glpi\vendor\psr\simple-cache\src)
psr/cache in (C:\inetpub\glpi\vendor\psr\cache\src)
league/csv in (C:\inetpub\glpi\vendor\league\csv\src)
mexitek/phpcolors in (C:\inetpub\glpi\vendor\mexitek\phpcolors\src\Mexitek\PHPColors)
guzzlehttp/guzzle in (C:\inetpub\glpi\vendor\guzzlehttp\guzzle\src)
guzzlehttp/psr7 in (C:\inetpub\glpi\vendor\guzzlehttp\psr7\src)
glpi-project/inventory_format in (C:\inetpub\glpi\vendor\glpi-project\inventory_format\lib\php)
wapmorgan/unified-archive in (C:\inetpub\glpi\vendor\wapmorgan\unified-archive\src)
paragonie/sodium_compat in (C:\inetpub\glpi\vendor\paragonie\sodium_compat\src)
symfony/cache in (C:\inetpub\glpi\vendor\symfony\cache)
html2text/html2text in (C:\inetpub\glpi\vendor\html2text\html2text\src)
symfony/css-selector in (C:\inetpub\glpi\vendor\symfony\css-selector)
symfony/dom-crawler in (C:\inetpub\glpi\vendor\symfony\dom-crawler)
twig/twig in (C:\inetpub\glpi\vendor\twig\twig\src)
twig/string-extra in (C:\inetpub\glpi\vendor\twig\string-extra)
symfony/polyfill-ctype not found
symfony/polyfill-iconv not found
symfony/polyfill-mbstring not found
symfony/polyfill-php80 in (C:\inetpub\glpi\vendor\symfony\polyfill-php80)
symfony/polyfill-php81 in (C:\inetpub\glpi\vendor\symfony\polyfill-php81)
symfony/polyfill-php82 in (C:\inetpub\glpi\vendor\symfony\polyfill-php82)

</details><details><summary>LDAP directories</summary><pre> 
Server: '192.168.0.1', Port: '389', BaseDN: 'dc=example;dc=com', Connection filter:
		'(&(objectClass=user)(objectCategory=person))', RootDN: 'example\administrador', Use TLS: none

</details><details><summary>SQL replicas</summary><pre> 
Not active

</details><details><summary>Notifications</summary><pre> 
Way of sending emails: SMTP (cpd@mail.example.com)

</details><details><summary>Plugins list</summary><pre> 
	actualtime           Name: ActualTime                     Version: 1.2.0      State: Not installed                           
		Install Method: Manual
	news                 Name: Alertas                        Version: 1.3.5      State: Installed / not activated               
		Install Method: Manual
	behaviors            Name: Comportamental                 Version: 2.0.2      State: Installed / not activated               
		Install Method: Manual
	mod                  Name: GLPI Modifications             Version: 1.1.9      State: Installed / not activated               
		Install Method: Manual
	resources            Name: Human Resources                Version: 2.4.3      State: Installed / not activated               
		Install Method: Manual
	pdf                  Name: Imprimir em PDF                Version: 1.3.1.1    State: Installed / not activated               
		Install Method: Manual
	timelineticket       Name: Linha do tempo dos chamados    Version: 9.2+1.0    State: Installed / not activated               
		Install Method: Manual
	mailanalyzer         Name: Mail Analyzer                  Version: 1.4.3      State: Installed / not activated               
		Install Method: Manual
	mobile               Name: Mobile                         Version: 1.1.6      State: Installed / not activated               
		Install Method: Manual
	archires             Name: Network Architectures          Version: 2.5.1      State: Installed / not activated               
		Install Method: Manual
	browsernotification  Name: Notificação do navegador       Version: 1.1.9      State: Installed / not activated               
		Install Method: Manual
	ocsinventoryng       Name: OCS Inventory NG               Version: 1.4.3      State: Installed / not activated               
		Install Method: Manual
	dashboard            Name: Painel                         Version: 0.9.3      State: Installed / not activated               
		Install Method: Manual
	reports              Name: Relatórios                     Version: 1.11.3     State: Installed / not activated               
		Install Method: Manual

</pre></details>
````

### Anything else?

Please let me know if there's any information that I can provide to troubleshoot/fix this notice.

**Comment by cedric-anne** [comment]
Hi,

Could you please provide the full stack trace ?

**Comment by eduardomozart** [comment]
Hello @cedric-anne,
Thank you for your fast response. I could not found anything at the GLPI PHP error log (``php-errors.log``) neither GLPI SQL error log (``sql-errors.log``). There's any location I can look for these?
Regards.

**Comment by cedric-anne** [comment]
In your `php-errors.log`, line `[21-Nov-2022 08:27:50 America/So_Paulo] PHP Notice: Undefined variable: username in C:\inetpub\glpi\src\Reservation.php on line 559` should be followed by some useful information.

**Comment by eduardomozart** [comment]
I'll try to enable PHP debug logs to see if anymore information are shown. But only this information is being shown right now.
I'm not sure if it's related to this issue, but into the "Simplified interface", isn't being shown the username of the reservation (as can be seen at the screenshot below) - it was working at the GLPI 9.2.4.

![WhatsApp Image 2022-11-21 at 09 26 15](https://user-images.githubusercontent.com/2974895/203056608-dcf7e666-613b-4036-be13-e6316741822e.jpeg)

After upgrading for GLPI 10.0.5, only Super-Admin members can see this information:

![WhatsApp Image 2022-11-21 at 09 26 55](https://user-images.githubusercontent.com/2974895/203056639-2dcc885f-5662-41ad-92c0-1f7907a8c0af.jpeg)

**Comment by eduardomozart** [comment]
There was the following commit some months ago at the Reservation about the user name, maybe it's related to this issue? https://github.com/glpi-project/glpi/pull/11994

**Comment by cedric-anne** [comment]
Fixed by #13445

**Comment by eduardomozart** [comment]
Hello @cedric-anne,
Thank you for your fix. From what I've seen, this fix simply removes the comment. But what I'd would like to confirm with you is the behavior of comments visibility. Into GLPI 9.2.4, the comments always shown at the Self-Service interface. It didn't seem to be the case anymore. It was useful because any user with access to the reservation screen could see who made the reservation and contact them. But with GLPI 10, we had to assign all our users as Technicians so they could access the Central UI and could see this information. From what I've seen, it happens because the comment is added only if the reservation is made from the user itself (``$myitem``) or if the user passes the boolean ``$canedit_admin`` (requires Central UI and read rights to the reservation). Why the behavior was changed and now comments information is being restricted to the Central UI only?
